docs/searching.md

```diff
@@ -249,4 +249,208 @@ var query = searcher.CreateQuery();
+```
+
+### String Facets
+
+String facets allows for counting the documents that share the same string value. This type of faceting is possible on all faceted index type.
+
+Basic example
+```csharp
+var searcher = myIndex.Searcher;
+var results = searcher.CreateQuery()
+ .Field("Address", "Hills")
+ .WithFacet("Address") // Get facets of the Address field
+ .Execute();
+
+var addressFacetResults = results.GetFacet("Address"); // Returns the facets for the specific field Address
+
+/* 
+* Example value
+* Label: Hills, Value: 2
+* Label: Hollywood, Value: 10
+*/
+
+var hillsValue = addressFacetResults.WithFacet("Hills"); // Gets the IFacetValue for the facet Hills
+```
+
+Filtered value example
+```csharp
+var searcher = myIndex.Searcher;
+var results = searcher.CreateQuery()
+ .Field("Address", "Hills")
+ .WithFacet("Address", "Hills") // Get facets of the Address field
+ .Execute();
+
+var addressFacetResults = results.GetFacet("Address"); // Returns the facets for the specific field Address
+
+/* 
+* Example value
+* Label: Hills, Value: 2 <-- As Hills was the only filtered value we will only get this facet
+*/
+
+var hillsValue = addressFacetResults.Facet("Hills"); // Gets the IFacetValue for the facet Hills
+```
+
+MaxCount example
+```csharp
+var searcher = myIndex.Searcher;
+var results = searcher.CreateQuery()
+ .Field("Address", "Hills")
+ .WithFacet("Address") // Get facets of the Address field
+ .Execute();
+
+var addressFacetResults = results.GetFacet("Address"); // Returns the facets for the specific field Address
+
+/* 
+* Example value
+* Label: Hills, Value: 2
+* Label: Hollywood, Value: 10
+* Label: London, Value: 12
+*/
+
+results = searcher.CreateQuery()
+ .Field("Address", "Hills")
+ .WithFacet("Address") // Get facets of the Address field
+    .MaxCount(2) // Gets the top 2 results (The facets with the highest value)
+ .Execute();
+
+addressFacetResults = results.GetFacet("Address"); // Returns the facets for the specific field Address
+
+/* 
+* Example value (Notice only 2 values are present)
+* Label: Hollywood, Value: 10
+* Label: London, Value: 12
+*/
+```
+
+FacetField example
+```csharp
+// Setup
+
+// Create a config
+var facetsConfig = new FacetsConfig();
+
+// Set the index field name to facet_address. This will store facets of this field under facet_address instead of the default $facets. This requires you to use FacetField in your Facet query. (Only works on string facets).
+facetsConfig.SetIndexFieldName("Address", "facet_address");
+
+services.AddExamineLuceneIndex("MyIndex",
+    // Set the indexing of your fields to use the facet type
+    fieldDefinitions: new FieldDefinitionCollection(
+        new FieldDefinition("Address", FieldDefinitionTypes.FacetFullText)
+        ),
+    // Pass your config
+    facetsConfig: facetsConfig
+    );
+
+
+var searcher = myIndex.Searcher;
+var results = searcher.CreateQuery()
+ .Field("Address", "Hills")
+ .WithFacet("Address") // Get facets of the Address field
```

Comment:
If you mean `WithFacet(string field, string facetField)` then this would overlap with the `WithFacet(string field, params string values)` and I think for most people using a specific facet field isn't necessary. This is a more advanced feature because you need to set the facet field via the `FacetsConfig`.

Reply:
If you need to specify the facet field in config, what is the reason you can also specify it in this API? (sorry, just trying to understand the process)

Reply:
By default all facet fields are stored in `$facets` then it's possible with the `FacetsConfig` in the index options (`LuceneIndexOptions`) to set another field name to store faceting data. Therefore if you index with a `FacetsConfig` setting the field name you need to know the facet field name when querying too. This means that if you query data with the default field name `$facets` but have changed it via config it will return an error from Lucene because `$facets` is not indexed.

I guess a solution would be to inject the `LuceneIndexOptions.FacetsConfig` into the LuceneSearchQuery and then check the config data for the facet field.

Reply:
Okay, I took a swing at injecting the `FacetsConfig` and it was almost painless. Unfortunately, this makes the searchers require the `FacetsConfig` as a parameter in the constructor. However, this shouldn't be a problem for anyone as you can always inject the `LuceneOptions` or take the lazy path and create a new config.

Reply:
This also makes `FacetField` immutable so that's nice